test/unit/MatchingEngine.spec.ts

```diff
@@ -150,96 +159,180 @@ describe('MatchingEngine.splitOrderByQuantity', () => {
 });
 
 describe('MatchingEngine.match', () => {
+  let engine: MatchingEngine;
+
+  beforeEach(() => {
+    engine = new MatchingEngine(loggers.orderbook, PAIR_ID);
+    isEmpty(engine);
+  });
+
   it('should fully match with two maker orders', () => {
-    const engine = new MatchingEngine(loggers.orderbook, PAIR_ID);
     engine.addPeerOrder(createPeerOrder(5, -5));
     engine.addPeerOrder(createPeerOrder(5, -5));
-    const matchAgainst = [engine.sellOrders];
-    const { remainingOrder } = engine.match(createOwnOrder(5, 10));
+    const { remainingOrder } = engine['match'](createOwnOrder(5, 10));
     expect(remainingOrder).to.be.undefined;
   });
 
   it('should split taker order when makers are insufficient', () => {
-    const engine = new MatchingEngine(loggers.orderbook, PAIR_ID);
     engine.addPeerOrder(createPeerOrder(5, -4));
     engine.addPeerOrder(createPeerOrder(5, -5));
-    const matchAgainst = [engine.sellOrders];
-    const { remainingOrder } = engine.match(createOwnOrder(5, 10));
+    const matchAgainst = [engine.queues.sell];
+    const { remainingOrder } = engine['match'](createOwnOrder(5, 10));
     expect(remainingOrder).to.not.be.undefined;
     expect(remainingOrder!.quantity).to.equal(1);
   });
 
   it('should split one maker order when taker is insufficient', () => {
-    const engine = new MatchingEngine(loggers.orderbook, PAIR_ID);
     engine.addPeerOrder(createPeerOrder(5, -5));
     engine.addPeerOrder(createPeerOrder(5, -6));
-    const matchAgainst = [engine.sellOrders];
-    const { matches, remainingOrder } = engine.match(createOwnOrder(5, 10));
+    const matchAgainst = [engine.queues.sell];
+    const { matches, remainingOrder } = engine['match'](createOwnOrder(5, 10));
     expect(remainingOrder).to.be.undefined;
     matches.forEach((match) => {
       expect(match.maker.quantity).to.equal(-5);
     });
-    const peekResult = engine.sellOrders.peek();
+    const peekResult = engine.queues.sell.peek();
     expect(peekResult).to.not.be.undefined;
     expect(peekResult!.quantity).to.equal(-1);
   });
 });
 
 describe('MatchingEngine.removeOwnOrder', () => {
-  it('should add a new ownOrder and then remove it', async () => {
-    const engine = new MatchingEngine(loggers.orderbook, PAIR_ID);
-    expect(engine.isEmpty()).to.be.true;
+  let engine: MatchingEngine;
 
+  beforeEach(() => {
+    engine = new MatchingEngine(loggers.orderbook, PAIR_ID);
+    isEmpty(engine);
```

Comment:
We have this exact code four times in this file.

Reply:
I can't put `beforeEach` on the global scope because it will be called on all the tests in the current run (all files), but I can do it for the other lines.